src/main/java/core/basesyntax/service/impl/ReportServiceImpl.java

```diff
@@ -0,0 +1,23 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.service.ReportService;
+import java.util.Map;
+
+public class ReportServiceImpl implements ReportService {
+    private static final String SEPARATOR = ",";
+
+    @Override
+    public String createFruitBalanceReport(Map<String, Integer> fruitBalance) {
+        StringBuilder reportBuilder = new StringBuilder("fruit")
+                .append(SEPARATOR).append("quantity").append(System.lineSeparator());
+
+        for (Map.Entry<String, Integer> balance : fruitBalance.entrySet()) {
+            if (balance.getValue() < 0) {
+                throw new RuntimeException("Fruit quantity can`t be less than 0");

```

Comment:
IllegalAurgumentException is more precise and suitable

Reply:
I сan`t replace RuntimeException with IllegalAccessException, because I need to write an error in the method signature through "throws" and make changes to the interface, which is not allowed by Dependency Inversion